spanner/api/SpannerTest/Tests.cs

```diff
@@ -133,13 +133,82 @@ public SpannerTests(SpannerFixture fixture)
                 if (!_fixture.s_initializedDatabase)
                 {
                     _fixture.s_initializedDatabase = true;
+                    DeleteStaleBackupsAndDatabases();
                     InitializeInstance();
                     InitializeDatabase();
                     InitializeBackup();
                 }
             }
         }
 
+        void DeleteStaleBackupsAndDatabases()
+        {
+            DatabaseAdminClient databaseAdminClient = DatabaseAdminClient.Create();
+            var instanceName = InstanceName.FromProjectInstance(_fixture.ProjectId, _fixture.InstanceId);
+            var databases = databaseAdminClient.ListDatabases(instanceName).Where(c => c.DatabaseName.DatabaseId.StartsWith("my-db-")).ToList();
+            var databasesToDelete = new List<string>();
+
+            // Delete all the databases created before 4 hrs.
+            var timestamp = DateTime.UtcNow.AddHours(-4).ToTimestamp().Seconds;
+            databases.ForEach(database =>
+            {
+                var databaseId = database.DatabaseName.DatabaseId.Replace("my-db-", "");
+                if (long.TryParse(databaseId, out long dbCreationTime) && dbCreationTime <= timestamp)
+                {
+                    databasesToDelete.Add(database.DatabaseName.DatabaseId);
+                }
+            });
+
+            Console.Out.WriteLineAsync($"{databasesToDelete.Count} old databases found.");
+
+            // Get backups.
+            ListBackupsRequest request = new ListBackupsRequest
+            {
+                ParentAsInstanceName = instanceName,
+                Filter = $"database:my-db-"
+            };
+            var backups = databaseAdminClient.ListBackups(request).ToList();
+
+            // Backups that belong to the databases to be deleted.
+            var backupsToDelete = backups.Where(c => databasesToDelete.Contains(DatabaseName.Parse(c.Database).DatabaseId)).ToList();
+
+            Console.Out.WriteLineAsync($"{backupsToDelete.Count} old backups found.");
+
+            // Delete the backups.
+            backupsToDelete.ForEach(backup =>
+            {
+                try
+                {
+                    DeleteBackup.SpannerDeleteBackup(_fixture.ProjectId, _fixture.InstanceId, backup.BackupName.BackupId);
+                }
+                catch (Exception) { }
+            });
+
+            // Delete the databases.
+            databasesToDelete.ForEach(databaseId =>
+            {
+                try
+                {
+                    _spannerCmd.Run("deleteDatabase", _fixture.ProjectId, _fixture.InstanceId, databaseId);
+                }
+                catch (Exception) { }
+            });
+        }
+
+        void InitializeInstance()
+        {
+            InstanceAdminClient instanceAdminClient = InstanceAdminClient.Create();
+            try
+            {
+                string name = $"projects/{_fixture.ProjectId}/instances/{_fixture.InstanceId}";

```

Comment:
There's probably an InstanceName class or there should be one. If it's missing, the protos are missing an annotation.